src/Endpoints/class-base-endpoint.php

```diff
@@ -156,4 +159,37 @@ public function register_endpoint(
 
 		register_rest_route( 'wp-parsely/v1', $endpoint, $rest_route_args );
 	}
+
+	/**
+	 * Registers the endpoint's WP REST route with arguments.
+	 *
+	 * @since 3.16.0
+	 *
+	 * @param string                      $endpoint The endpoint's route.
+	 * @param string                      $callback The callback function to call when the endpoint is hit.
+	 * @param array<string>               $methods The HTTP methods to allow for the endpoint.
+	 * @param array<WP_HTTP_Request_Args> $args The arguments for the endpoint.
+	 */
+	public function register_endpoint_with_args(
```

Comment:
The idea is that this function will allow to register endpoint with custom item args, so it could process and validate the common API properties (such as the post ID) directly in the endpoint, instead of doing on the callback.

I thought about adding it to the `register_endpoint` function, but it would require changing the function signature and validating all the instances where it is being used. However, your suggestion it's something that I think it makes sense to do, and I will try to have a better solution in the next PR that focuses on the backend. This part of the code can be considered WIP.

Reply:
No issue if you feel it's better to have an additional function. Thank you for the explanation!